staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager_test.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fieldmanager_test
+
+import (
+	"errors"
+	"testing"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager"
+)
+
+type fakeObjectConvertor struct{}
+
+func (c *fakeObjectConvertor) Convert(in, out, context interface{}) error {
+	out = in
+	return nil
+}
+
+func (c *fakeObjectConvertor) ConvertToVersion(in runtime.Object, _ runtime.GroupVersioner) (runtime.Object, error) {
+	return in, nil
+}
+
+func (c *fakeObjectConvertor) ConvertFieldLabel(_ schema.GroupVersionKind, _, _ string) (string, string, error) {
+	return "", "", errors.New("not implemented")
+}
+
+type fakeObjectDefaulter struct{}
+
+func (d *fakeObjectDefaulter) Default(in runtime.Object) {}
+
+func NewTestFieldManager(t *testing.T) *fieldmanager.FieldManager {
+	gv := schema.GroupVersion{
+		Group:   "apps",
+		Version: "v1",
+	}
+
+	return fieldmanager.NewCRDFieldManager(
+		&fakeObjectConvertor{},
+		&fakeObjectDefaulter{},
+		gv,
+		gv,
+	)
+}
+
+func TestFieldManagerCreation(t *testing.T) {
+	if NewTestFieldManager(t) == nil {
+		t.Fatal("failed to create FieldManager")
+	}
+}
+
+func TestApplyStripsFields(t *testing.T) {
+	f := NewTestFieldManager(t)
+
+	obj := &corev1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:              "a",
+			Namespace:         "a",
+			CreationTimestamp: metav1.Time{Time: time.Time{}},
+			SelfLink:          "a",
+		},
+	}
+
+	newObj, err := f.Apply(obj, []byte(`{
+		"apiVersion": "v1",
+		"kind": "Deployment",
+		"metadata": {
+			"name": "b",
```

Comment:
That really shouldn't happen though (applying with a different name/namespace), it might not matter in this test but that's confusing.

Reply:
Sure, it's my only way of testing if the fields get stripped on changes. Or is there another one?